lib/phlex/rails/streaming.rb

```diff
@@ -17,10 +17,36 @@ def stream(view)
 		self.response_body = Enumerator.new do |buffer|
 			view.call(buffer, view_context: view_context)
 		rescue => e
-			buffer << %('">)
-			buffer << view_context.javascript_tag(nonce: true) { %(window.location = "/500.html").html_safe }
+			raise (e) if Rails.env.test?
 
-			raise e
+			debug_middleware = ActionDispatch::DebugExceptions.new(
+				proc { |env| raise(e) },
+				response_format: :html
+			)
+
+			_debug_status, _debug_headers, debug_body = debug_middleware.call(request.env)
+
+			if Rails.env.development?
+				js = <<~JAVASCRIPT
+					document.documentElement.innerHTML = "#{view_context.j(debug_body.join)}";
+
+					// Re-evaluate all script tags
+					document.querySelectorAll("script").forEach((script) => {
+						newScript = document.createElement("script");
+						newScript.text = script.text;
+						script.replaceWith(newScript);
+					});
+				JAVASCRIPT
+			else
+				js = <<~JAVASCRIPT
+					window.location = "/500.html";
+				JAVASCRIPT
+			end
+
+			buffer << %(-->"'>)
+			buffer << view_context.javascript_tag(js, nonce: true)
+
+			p e
```

Comment:
This seems left over from local testing?

Reply:
No, it’s so you can see the error in the console. We can probably improve this somewhat, but we should have some useful output in the console as well as the page.

Reply:
We can’t use `raise` it again, because it blocks the request. Unless there’s some way to manually tell ActionDispatch to close the request.